### PR TITLE
[feature-#11636] set ip by specifying

### LINKED ISSUE #11636: [Feature][Worker] Get ip by specifying
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

## One network card has two ips
displayName of eth1:
```
eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 52:54:00:75:0d:7a brd ff:ff:ff:ff:ff:ff
    inet 172.31.252.109/23 scope global eth1
       valid_lft forever preferred_lft forever
    inet 172.16.98.30/24 scope global eth1
       valid_lft forever preferred_lft forever
```

### Use case

- i want to specify ip
- update codes in method of getLocalAddress0

```java

  private static synchronized InetAddress getLocalAddress0() {
        if (null != LOCAL_ADDRESS) {
            return LOCAL_ADDRESS;
        }

        InetAddress localAddress = null;
        String networkIp = PropertyUtils.getString(Constants.DOLPHIN_SCHEDULER_NETWORK_IP);
        if(StringUtils.isNotBlank(networkIp)){
            try {
                localAddress =  InetAddress.getByName(networkIp);
                return localAddress;
            } catch (UnknownHostException e) {
                e.printStackTrace();
            }
        }
```

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuchanghai**
## Purpose of the pull request

set ip by specifying

## Verify this pull request

This pull request is code cleanup without any test coverage.

this close #11636

### CODE REVIEW

File: dolphinscheduler-common/src/main/resources/common.properties [hunk 1]
```diff
@@ -92,6 +92,9 @@ sudo.enable=true
 # network IP gets priority, default: inner outer
 #dolphin.scheduler.network.priority.strategy=default
 
+# specify network IP . example:127.0.0.1, default:empty
```
Comment (SbloodyS) [inline dolphinscheduler-common/src/main/resources/common.properties]:
```suggestion
# specify network IP. example: 127.0.0.1, default: empty
```

File: docs/docs/zh/guide/resource/configuration.md [hunk 2]
```diff
@@ -123,6 +123,9 @@ sudo.enable=true
 # network IP gets priority, default: inner outer
 #dolphin.scheduler.network.priority.strategy=default
 
+# specify network IP . example:127.0.0.1, default:empty
```
Comment (SbloodyS) [inline docs/docs/zh/guide/resource/configuration.md]:
```suggestion
# specify network IP. example: 127.0.0.1, default: empty
```

File: docs/docs/en/guide/resource/configuration.md [hunk 3]
```diff
@@ -116,6 +116,9 @@ sudo.enable=true
 # network IP gets priority, default: inner outer
 #dolphin.scheduler.network.priority.strategy=default
 
+# specify network IP . example:127.0.0.1, default:empty
```
Comment (SbloodyS) [inline docs/docs/en/guide/resource/configuration.md]:
```suggestion
# specify network IP. example: 127.0.0.1, default: empty
```

File: deploy/kubernetes/dolphinscheduler/values.yaml [hunk 4]
```diff
@@ -135,6 +135,9 @@ conf:
     # network IP gets priority, default: inner outer
     #dolphin.scheduler.network.priority.strategy: default
 
+    # specify network IP . example:127.0.0.1, default:empty
```
Comment (SbloodyS) [inline deploy/kubernetes/dolphinscheduler/values.yaml]:
```suggestion
    # specify network IP. example: 127.0.0.1, default: empty
```

**Comment by fuchanghai** [comment]
@SbloodyS Thanks for your suggestions

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/NetUtils.java [hunk 5]
```diff
@@ -118,9 +119,12 @@ private static synchronized InetAddress getLocalAddress0() {
         if (null != LOCAL_ADDRESS) {
             return LOCAL_ADDRESS;
         }
-
+        String specifyNetworkIP = PropertyUtils.getString(Constants.DOLPHIN_SCHEDULER_NETWORK_IP);
```
Comment (caishunfeng) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/NetUtils.java:122]: It's not a good way to add specify ip logic in method `getLocalAddress0`.

**Review by caishunfeng — CHANGES_REQUESTED** [review]

**Comment by caishunfeng** [comment]
Hi @fuchanghai , I think what you want specify ip is just for the local task like shell、python can be executed in another merchine, is it right?

If so, what's the failover logic if worker down?

**Comment by fuchanghai** [comment]
@caishunfeng My current situation is that there is a network card with dual IPs, one of which can communicate with me locally, and the other cannot. If you do not specify ip in this method, if you obtain it in other methods, you can only obtain the corresponding network card, and through this network card, you may obtain an IP that cannot be connected. When I get the wrong IP, I can't connect to the wrong IP locally, because it is not in a switch, which will cause the log to fail to query

**Comment by caishunfeng** [comment]
> @caishunfeng My current situation is that there is a network card with dual IPs, one of which can communicate with me locally, and the other cannot. If you do not specify ip in this method, if you obtain it in other methods, you can only obtain the corresponding network card, and through this network card, you may obtain an IP that cannot be connected. When I get the wrong IP, I can't connect to the wrong IP locally, because it is not in a switch, which will cause the log to fail to query

OK, I get your point now. So the specify worker ip is just designed for more than one ip in a network card. 
But I think this is a problem of network setting, not the problem for ds. There should be other better ways to solve it, such as using Intranet host instead of IP, WDYT?

**Comment by fuchanghai** [comment]
> > @caishunfeng My current situation is that there is a network card with dual IPs, one of which can communicate with me locally, and the other cannot. If you do not specify ip in this method, if you obtain it in other methods, you can only obtain the corresponding network card, and through this network card, you may obtain an IP that cannot be connected. When I get the wrong IP, I can't connect to the wrong IP locally, because it is not in a switch, which will cause the log to fail to query
> 
> OK, I get your point now. So the specify worker ip is just designed for more than one ip in a network card. But I think this is a problem of network setting, not the problem for ds. There should be other better ways to solve it, such as using Intranet host instead of IP, WDYT?

In the case I just mentioned, the two different IP addresses of the same network card are both intranet addresses, but one is on the same switch as my local computer, and the other is not. So one can connect, but the other cannot.

**Comment by fuchanghai** [comment]
> > @caishunfeng My current situation is that there is a network card with dual IPs, one of which can communicate with me locally, and the other cannot. If you do not specify ip in this method, if you obtain it in other methods, you can only obtain the corresponding network card, and through this network card, you may obtain an IP that cannot be connected. When I get the wrong IP, I can't connect to the wrong IP locally, because it is not in a switch, which will cause the log to fail to query
> 
> OK, I get your point now. So the specify worker ip is just designed for more than one ip in a network card. But I think this is a problem of network setting, not the problem for ds. There should be other better ways to solve it, such as using Intranet host instead of IP, WDYT?

In the case I just mentioned, the two different IP addresses of the same network card are both intranet addresses, but one is on the same switch as my local computer, and the other is not. So one can connect, but the other cannot.

**Comment by fuchanghai** [comment]
> > @caishunfeng My current situation is that there is a network card with dual IPs, one of which can communicate with me locally, and the other cannot. If you do not specify ip in this method, if you obtain it in other methods, you can only obtain the corresponding network card, and through this network card, you may obtain an IP that cannot be connected. When I get the wrong IP, I can't connect to the wrong IP locally, because it is not in a switch, which will cause the log to fail to query
> 
> OK, I get your point now. So the specify worker ip is just designed for more than one ip in a network card. But I think this is a problem of network setting, not the problem for ds. There should be other better ways to solve it, such as using Intranet host instead of IP, WDYT?

Two IPs on the same network card cannot communicate with each other

**Comment by caishunfeng** [comment]
> > > @caishunfeng My current situation is that there is a network card with dual IPs, one of which can communicate with me locally, and the other cannot. If you do not specify ip in this method, if you obtain it in other methods, you can only obtain the corresponding network card, and through this network card, you may obtain an IP that cannot be connected. When I get the wrong IP, I can't connect to the wrong IP locally, because it is not in a switch, which will cause the log to fail to query
> > 
> > 
> > OK, I get your point now. So the specify worker ip is just designed for more than one ip in a network card. But I think this is a problem of network setting, not the problem for ds. There should be other better ways to solve it, such as using Intranet host instead of IP, WDYT?
> 
> In the case I just mentioned, the two different IP addresses of the same network card are both intranet addresses, but one is on the same switch as my local computer, and the other is not. So one can connect, but the other cannot.

WDYT @SbloodyS @ruanwenjun @zhongjiajie

**Comment by zhongjiajie** [comment]
> > > > @caishunfeng My current situation is that there is a network card with dual IPs, one of which can communicate with me locally, and the other cannot. If you do not specify ip in this method, if you obtain it in other methods, you can only obtain the corresponding network card, and through this network card, you may obtain an IP that cannot be connected. When I get the wrong IP, I can't connect to the wrong IP locally, because it is not in a switch, which will cause the log to fail to query
> > > 
> > > 
> > > OK, I get your point now. So the specify worker ip is just designed for more than one ip in a network card. But I think this is a problem of network setting, not the problem for ds. There should be other better ways to solve it, such as using Intranet host instead of IP, WDYT?
> > 
> > 
> > In the case I just mentioned, the two different IP addresses of the same network card are both intranet addresses, but one is on the same switch as my local computer, and the other is not. So one can connect, but the other cannot.
> 
> WDYT @SbloodyS @ruanwenjun @zhongjiajie

Do we have another way to handle this situation? I think all IP should work in network card, and should find another way to handle your user-case to connect to your local.

**Comment by zhongjiajie** [comment]
BTW, if we have to add this configuration, please add it to method `getLocalAddress` instead of `getLocalAddress0`

**Comment by fuchanghai** [comment]
@zhongjiajie ok,I will continue to discuss the feasibility with our operations engineers

**Pull request closed without merging**